### PR TITLE
Boolean types and Numbers with commas

### PR DESCRIPTION
**Pull request by davidfig**
- [x] parse numbers, currency, and percentages with commas
- [x] rust and TS support for boolean values
- [x] render boolean
- [x] also fixed center alignment (it was off by a couple pixels) 

Note: Python receives `TRUE` and `FALSE` as a string. This will be fixed with the Python types work.
 
Screenshot 2024-02-05 at 5 47 23 AM

### CODE REVIEW

File: quadratic-core/src/controller/operations/cell_value.rs [hunk 1]
```diff
@@ -47,7 +53,15 @@ impl GridController {
                 }
             }
             CellValue::Number(number)
-        } else if let Ok(bd) = BigDecimal::from_str(value) {
+        } else if let Some(bool) = CellValue::unpack_boolean(value) {
+            bool
+        } else if let Ok(bd) = BigDecimal::from_str(&CellValue::strip_commas(value)) {
+            if value.contains(',') {
```
Comment (ddimaria) [inline quadratic-core/src/controller/operations/cell_value.rs:59]: Duplication here, consider abstracting into a function to call in both places.

File: quadratic-client/src/gridGL/cells/cellsLabel/CellLabel.ts [hunk 2]
```diff
@@ -94,6 +94,10 @@ export class CellLabel extends Container {
     } else {
       this.tint = 0;
     }
+    if (cell.special === 'True' || cell.special === 'False') {
```
Comment (ddimaria) [inline quadratic-client/src/gridGL/cells/cellsLabel/CellLabel.ts:97]: Is this case guaranteed?

File: quadratic-core/src/grid/sheet/rendering.rs [hunk 3]
```diff
@@ -90,25 +90,34 @@ impl Sheet {
...
+                        });
+                        let numeric_decimals = self.decimal_places(Pos { x, y }, is_percentage);
+                        let numeric_commas = column.numeric_commas.get(y);
+
+                        // if align is not set, set it to right only for numbers
+                        align = align.or(Some(CellAlign::Right));
+
+                        value.to_display(numeric_format, numeric_decimals, numeric_commas)
+                    }
+                    CellValue::Logical(bool) => {
```
Comment (ddimaria) [inline quadratic-core/src/grid/sheet/rendering.rs:113]: Possibly implement: `impl From<bool> for CellValue`

File: quadratic-core/src/values/cellvalue.rs [hunk 4]
```diff
@@ -245,18 +245,31 @@ impl CellValue {
         None
     }
 
+    pub fn unpack_boolean(s: &str) -> Option<CellValue> {
+        match s.to_ascii_lowercase().as_str() {
```
Comment (ddimaria) [inline quadratic-core/src/values/cellvalue.rs:249]: Possibly implement in: `impl From<&str> for CellValue`  (I know we have some of that written already)

File: quadratic-core/src/values/cellvalue.rs [hunk 5]
```diff
@@ -436,9 +453,22 @@ impl CellValue {
                     attr: CellFmtArray::NumericDecimals(RunLengthEncoding::repeat(Some(2), 1)),
                 });
             }
-        } else if let Ok(bd) = BigDecimal::from_str(s) {
+
+            if s.contains(',') {
```
Comment (ddimaria) [inline quadratic-core/src/values/cellvalue.rs:457]: Duplication again here

File: quadratic-core/src/values/cellvalue.rs [hunk 6]
```diff
@@ -436,9 +453,22 @@ impl CellValue {
...
+                ops.push(Operation::SetCellFormats {
+                    sheet_rect,
+                    attr: CellFmtArray::NumericCommas(RunLengthEncoding::repeat(Some(true), 1)),
+                });
+            }
+        } else if let Ok(bd) = BigDecimal::from_str(&CellValue::strip_commas(s)) {
+            if s.contains(',') {
```
Comment (ddimaria) [inline quadratic-core/src/values/cellvalue.rs:464]: Duplication again here

**Review by ddimaria — APPROVED** [review]
Some non-blocking comments

File: quadratic-client/src/gridGL/cells/cellsLabel/CellLabel.ts (hunk 2 above)
Comment (davidfig) [reply quadratic-client/src/gridGL/cells/cellsLabel/CellLabel.ts]: Yes. `True` and `False` map to Logical(true) and Logical(false) in TS.

**Pull request merged**